cellpylib/ca_functions.py

```diff
@@ -84,13 +84,21 @@ def binary_rule(neighbourhood, rule, scheme=None):
     the NKS convention). If 'nks' is provided for the scheme parameter, the NKS convention is used for listing the 
     neighbourhoods.
     :param neighbourhood: a binary array of length 2r + 1
-    :param rule: an int indicating the cellular automaton rule number
+    :param rule: an int or a binary array indicating the cellular automaton rule number
     :param scheme: can be None (default) or 'nks'; if 'nks' is given, the rule numbering scheme used in NKS is used
+    :param powers_of_two: a pre-computed array containing the powers of two; can be None (default) or an array of length
+     len(neighbourhood); if an array is given, it will used to speed up the calculation of state_int
```

Comment:
I notice that the powers of two, when provided, should be a list such as `[4,2,1]`. We should probably state this as an example in the docstring.

Reply:
Added your example to the docstring